src/Catalyst.Cli/Commands/GetFileCommand.cs

```diff
@@ -0,0 +1,91 @@
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System.Threading;
+using Catalyst.Abstractions.Cli.Commands;
+using Catalyst.Abstractions.FileTransfer;
+using Catalyst.Cli.CommandTypes;
+using Catalyst.Cli.Options;
+using Catalyst.Core.Lib.Extensions;
+using Catalyst.Core.Lib.FileTransfer;
+using Catalyst.Core.Lib.IO.Messaging.Dto;
+using Catalyst.Protocol.Rpc.Node;
+using Serilog;
+
+namespace Catalyst.Cli.Commands
+{
+    public sealed class GetFileCommand :
+        BaseMessageCommand<GetFileFromDfsRequest, GetFileFromDfsResponse, GetFileOptions>
+    {
+        private readonly IDownloadFileTransferFactory _downloadFileTransferFactory;
+
+        public GetFileCommand(IDownloadFileTransferFactory downloadFileTransferFactory,
+            ICommandContext commandContext,
+            ILogger logger) : base(commandContext, logger)
+        {
+            _downloadFileTransferFactory = downloadFileTransferFactory;
+        }
+
+        protected override GetFileFromDfsRequest GetMessage(GetFileOptions option)
+        {
+            return new GetFileFromDfsRequest
+            {
+                DfsHash = option.FileHash
+            };
+        }
+
+        public override void SendMessage(GetFileOptions opts)
+        {
+            var message = GetMessage(opts);
+            var protocolMessage = message.ToProtocolMessage(SenderPeerId);
+            var correlationId = protocolMessage.CorrelationId.ToCorrelationId();
+
+            var messageDto = new MessageDto(
+                protocolMessage,
+                RecipientPeerId);
+
+            var fileTransfer = new DownloadFileTransferInformation(
+                SenderPeerId,
+                RecipientPeerId,
+                Target.Channel,
+                correlationId,
+                opts.FileOutput,
+                0
+            );
+
+            _downloadFileTransferFactory.RegisterTransfer(fileTransfer);
+            Target.SendMessage(messageDto);
+
+            while (!fileTransfer.ChunkIndicatorsTrue() && !fileTransfer.IsExpired())
+            {
+                CommandContext.UserOutput.Write($"\rDownloaded: {fileTransfer.GetPercentage().ToString()}%");
+                Thread.Sleep(500);
+            }
+
+            if (fileTransfer.ChunkIndicatorsTrue())
+                CommandContext.UserOutput.Write($"\rDownloaded: {fileTransfer.GetPercentage().ToString()}%\n");
+            else
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'else' block.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4594394)